src/components/calcite-modal/calcite-modal.tsx

```diff
@@ -93,42 +70,51 @@ export class CalciteModal {
         theme={theme}
       >
         <div class="modal">
-          <focus-trap ref={(el: FocusTrap) => (this.trap = el)}>
-            <div class="modal__header">
-              <button
-                class="modal__close"
-                aria-label={this.closeLabel}
-                onClick={() => this.close()}
-              >
-                <svg
-                  xmlns="http://www.w3.org/2000/svg"
-                  height="24"
-                  width="24"
-                  viewBox="0 0 24 24"
-                  fill="currentColor"
-                >
-                  <path d={x24} />
-                </svg>
-              </button>
-              <header class="modal__title">
-                <slot name="header"></slot>
-              </header>
-            </div>
-            <div class="modal__content">
-              <slot name="content"></slot>
-            </div>
-            <div
-              class={{
-                modal__footer: true,
-                "modal__footer--hide-back": this.hideBackButton,
-                "modal__footer--hide-secondary": this.hideSecondaryButton
-              }}
+          <div
+            data-focus-fence="true"
```

Comment:
What was wrong with the `focus-trap` component?

Reply:
it caused the entire component not to render in Edge... something about having a nested Shadow DOM set cause the layout to get screwed up but only in edge... I'm still using the base methods from that library, as they seem to work cross browser.

Reply:
Hm, that's unfortunate!

Is there anywhere else in calcite-components where we use focus trap? If so maybe it could be generalized to be a mixin or decorator or something.

Reply:
I looked for the focus trap element and this was the only spot, I think @macandcheese did some work on focus for dropdown elements, too?